accepted/0000-workspaces-part2.md

```diff
@@ -0,0 +1,221 @@
+> done
+
+> dep-b@1.3.1 test /Users/username/foo/dep-b
+> done
+```
+
+**NOTE:** `publish` and `version` are very special cases in a **npm workspaces** setup and are outside of the scope of this RFC. These features will live in userland in the form of popular modules such as [Lerna](https://lerna.js.org/) for the foreseeable future.
+
+### 2. Filter a subset of workspaces
+
+Filter is done via named argument (`--workspace`, short: `-w`) and here are some of the reasons why we decided to go that route:
+- [Lerna filters](https://www.npmjs.com/package/@lerna/filter-options) were the starting point but early in the discussion some other considerations were brought forward
+- npm configs do have some baggage, configs might apply to to other cli commands and can also possibly be defined project or system-wide - that said, there are things we can do to avoid these pitfalls (such as making sure `npm_config_workspace` env varibable never gets set in `run-script`)
+- During the many discussions around this RFC a few strong points for named arguments stand out:
+  - Allow us to keep `npm workspace` for other command usages
+  - npm users are more familiar with usage of named configs, example: `npm test --workspace=foo` is very similar to `npm test --prefix=./packages/foo` - if you think on it in terms of "what workspace am I in" again, a named config makes more sense in npm
+  - [Organized a poll](https://github.com/npm/rfcs/issues/160) in which named arguments were the leading option, here are the results for reference:
+  ```
+  1. (12 votes) 21%  -  Positional argument with one-char alias: `npm workspace foo test` or `npm w foo test`
+  2. (11 votes) 11%  -  Positional argument but shorter default name: `npm ws foo test` or `npm w foo test`
+  3. (13 votes) 13%  -  Special character (operator-like): `npm :foo test`
+  4. (20 votes) 35%  -  Named argument: `npm --workspace=foo test` or `npm -w=foo test`
+  ```
+
+Given the results of all this preliminar work, the preferred way to run a command in the context of a workspace is to use the named `--workspace` argument or its `-w` short alias, e.g:
+
+In a project with the following structure:
+```
+./
+├── package.json { "name": "root", "workspaces": ["packages/foo", "packages/bar"] }
+└── packages/
+    ├── foo/
+    │   └── package.json { "name": "foo", "version": "1.0.0" }
+    └── bar/
+        └── package.json { "name": "foo", "version": "1.0.0" }
+```
+
+You can run tests for the `foo` workspace from the root of your project, with the following syntax:
+
+```
+npm test --workspace=foo
+```
+
+
+#### Install dependency example:
+
+Add `tap` as a **dependency** of a **workspace** named `foo`:
+
+```sh
+npm install tap --workspace=foo
+```
+
+
+Note: **Globs** are not supported as a valid `--workspace` argument value, the alternative is to use `npm <cmd> --workspace=<alias>` in which the `alias` can be defined beforehand in your **workspace** configuration as detailed in the following up section.
```

Comment:
What about comma-separated values? EG `-w foo,bar`

Reply:
That's not very idiomatic/common for CLI args, in my experience.

Reply:
yeah, I've not really seen that before and would not expect it.

Reply:
As long as we've got aliases, this isn't an issue.

Reply:
the way [nopt](https://www.npmjs.com/package/nopt) works is that you specify multiple times that same item (and I don't have any intention to change from the default behavior here) so assuming we want to support multiple **workspaces** declarations the following should be valid: `npm <cmd> -w foo -w bar`

but @mikestopcontinues I think you got the idea of the design here in which I want to promote/steer usage of pre-defined aliases instead